src/plugins/saved_objects_management/public/management_section/objects_table/saved_objects_table.tsx

```diff
@@ -178,43 +178,62 @@ export class SavedObjectsTable extends Component<SavedObjectsTableProps, SavedOb
   }
 
   fetchCounts = async () => {
-    const { allowedTypes } = this.props;
+    const { allowedTypes, namespaceRegistry } = this.props;
     const { queryText, visibleTypes, visibleNamespaces } = parseQuery(this.state.activeQuery);
 
     const filteredTypes = allowedTypes.filter(
       (type) => !visibleTypes || visibleTypes.includes(type)
     );
 
+    let availableNamespaces = namespaceRegistry.getAll() || [];
+    availableNamespaces = availableNamespaces.map((ns) => ns.id);
+
+    const filteredCountOptions = {
+      typesToInclude: filteredTypes,
+      searchString: queryText,
+    };
+
+    if (availableNamespaces?.length) {
+      const filteredNamespaces = availableNamespaces.filter(
+        (ns) => !visibleNamespaces || visibleNamespaces.includes(ns)
```

Comment:
is `visibleNamespaces` same for all tenant? if not, how is the value determined?

Reply:
`visibleNamespaces` is the selected tenants from the new tenant filter. Its returned from `parseQuery`.